### ISSUE TITLE
String.create error in interpreter

### ISSUE THREAD

**Issue by kripken** [WebAssembly/spec#156]
I tried to parse the Poppler PDF library compiled to `.wast` in the interpreter. It's 39MB, and the interpreter errors on

```
[..]/spec/ml-proto/wasm: uncaught exception Invalid_argument("String.create")
Raised by primitive operation at file "host/main.ml", line 12, characters 2-206
Called from file "host/main.ml", line 53, characters 15-24
Called from file "list.ml", line 73, characters 12-15
Called from file "host/main.ml", line 86, characters 4-33
```

I'm not that familiar with ocaml, so I'm not sure what to do to investigate this further.

I can attach the file if that's useful (too big for github though, I think).

**Comment by kripken** [comment]
Investigating this, it looks like the interpreter has a limit of around 16MB on input size, any larger and that error is hit. There is nothing special in the content, it's just a size issue.

I know nothing about ocaml (I spent some time today trying to get a printf for debugging in the interpreter, unsuccessfully ;) ), is there a fixed string size limit there or something like that?

**Comment by hhugo** [comment]
There is indeed a 16mb limit for string in 32bit. Try with a 64 bit ocaml compiler.

**Comment by kripken** [comment]
I see, thanks. I'm on 32-bit atm so I guess I'm stuck.

**Comment by hhugo** [comment]
A solution would be not to load the all file as a string - by using `Lexing.from_channel` instead of `Lexing.from_string`.
https://github.com/WebAssembly/spec/blob/master/ml-proto/host/main.ml#L22

**Comment by rossberg-chromium** [comment]
I believe this was fixed by #197. Please reopen if you keep seeing the problem.
